core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4146,9 +4149,52 @@ private static List<Expression> args(RexCall call,
         if (argValueList.size() == 2) {
```

Comment:
can the function only have 1 or 2 arguments? If so, let's rewrite this as an `if (size == 1) ... else ...`

Reply:
done